pinot-common/src/main/java/org/apache/pinot/common/utils/helix/HelixHelper.java

```diff
@@ -163,6 +169,31 @@ public Boolean call() {
             return true;
           }
         }
+
+        private boolean shouldCompress(IdealState is) {
+          if (is.getNumPartitions() > NUM_PARTITIONS_THRESHOLD_TO_ENABLE_COMPRESSION) {
+            return true;
+          }
+
+          // Find the number of characters in one partition in idealstate, and extrapolate
+          // to estimate the number of characters.
+          Iterator<String> it = is.getPartitionSet().iterator();
```

Comment:
qq: can we just check the string's length of the ideal state? By doing so, even if the length is not purely from partitions the partition set, the overhead from the rest of the fields might not be that high.

Reply:
Also, according to the name from the variable (`MIN_NUM_CHARS_IN_IS_TO_TURN_ON_COMPRESSION`), it doesn't exclude the rest of the fields though.

Reply:
> qq: can we just check the string's length of the ideal state? By doing so, even if the length is not purely from partitions the partition set, the overhead from the rest of the fields might not be that high.

To check the length we will need to serialize it, I was trying to avoid that.

Reply:
> Also, according to the name from the variable (`MIN_NUM_CHARS_IN_IS_TO_TURN_ON_COMPRESSION`), it doesn't exclude the rest of the fields though.

That is correct.  I could not come up with another name though :)